frame/scheduler/src/lib.rs

```diff
@@ -15,38 +15,62 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! # Scheduler
-//! A Pallet for scheduling dispatches.
+//! # Scheduler Pallet
 //!
-//! - [`Config`]
-//! - [`Call`]
-//! - [`Pallet`]
+//! A Pallet for scheduling runtime calls.
 //!
 //! ## Overview
 //!
-//! This Pallet exposes capabilities for scheduling dispatches to occur at a
-//! specified block number or at a specified period. These scheduled dispatches
+//! This Pallet exposes capabilities for scheduling runtime calls to occur at a
+//! specified block number or at a specified period. These scheduled runtime calls
 //! may be named or anonymous and may be canceled.
 //!
-//! **NOTE:** The scheduled calls will be dispatched with the default filter
-//! for the origin: namely `frame_system::Config::BaseCallFilter` for all origin
-//! except root which will get no filter. And not the filter contained in origin
-//! use to call `fn schedule`.
+//! **NOTE:** Instead of the filter contained in the origin used to call `fn schedule`, scheduled runtime calls will be 
+//! dispatched with the default filter for the origin: namely `frame_system::Config::BaseCallFilter` for all origin
+//! types (except root which will get no filter).
 //!
 //! If a call is scheduled using proxy or whatever mecanism which adds filter,
-//! then those filter will not be used when dispatching the schedule call.
+//! then those filter will not be used when dispatching the schedule runtime call.
 //!
-//! ## Interface
+//! ## Pallet API
 //!
-//! ### Dispatchable Functions
+//! See the [`pallet`] module for more information about the interfaces this pallet exposes,
+//! including its configuration trait, dispatchables, storage items, events and errors.
 //!
-//! * `schedule` - schedule a dispatch, which may be periodic, to occur at a specified block and
-//!   with a specified priority.
-//! * `cancel` - cancel a scheduled dispatch, specified by block number and index.
-//! * `schedule_named` - augments the `schedule` interface with an additional `Vec<u8>` parameter
-//!   that can be used for identification.
-//! * `cancel_named` - the named complement to the cancel function.
-
+//! ### Warning
+//!
+//! This Pallet executes all scheduled runtime calls in the
+//! [`frame_support::traits::Hooks::on_initialize`] hook.
+//!
+//! **Do NOT schedule runtime calls if they satisfy any of the following points:**
+//!
+//! * [ ] Runtime call does not need to be executed at a specific block (not a hard requirement).
```

Comment:
I think what you saying here is more a property of `on_initialize` than this pallet. I would phrase is as this sequence: 

1. This pallet uses `on_initialize`
2. `on_initialize` has a mandatory property to it (which you can make sure is documented in the `Hooks` trait), and should only be used for _really_ mandatory stuff. There, you can also mention `on_idle`
4. Therefore, this pallet is also suitable for scheduling _really mandatory_ stuff. 


In other words, I don't think scheduler and `on_idle` are apples to apples.